scipy/special/ellint_carlson_cpp_lite/ellint_argcheck.hh

```diff
@@ -0,0 +1,105 @@
+#ifndef ELLINT_ARGCHECK_HH_INCLUDED
+#define ELLINT_ARGCHECK_HH_INCLUDED
+
+
+#include <complex>
+#include "ellint_typing.hh"
+#include "ellint_common.hh"
+
+
+namespace ellint_carlson { namespace argcheck
+{
+    template<typename T>
+    inline constexpr typing::real_only<T, bool>
+    invalid_rerr(const T& r, const double upper)
+    {
+	return ( r <= 0.0 ) || ( r > upper );
+    }
+
+
+    template<typename T>
+    inline typing::real_only<T, bool>
+    too_small(const T& x)
+    {
+	return ( (x == 0.0 ) || ( std::fpclassify(x) == FP_SUBNORMAL ) );
+    }
+
+    template<typename T>
+    inline typing::cplx_only<T, bool>
+    too_small(const T& z)
+    {
+	return ( too_small(z.real()) && too_small(z.imag()) );
+    }
+
+
+    /* Argument (phase) angle is NOT +/- pi, and is not indeterminate value,
+     * checked without computing the actual argument. */
+    template<typename T>
+    inline constexpr typing::real_only<T, bool>
+    ph_good(const T& x)
+    {
+	/* also handles x being NaN, which compares to false, hence is the
+	 * correct result. */
+	return ( x >= 0.0 );
+    }
+
+    template<typename T>
+    inline typing::cplx_only<T, bool>
+    ph_good(const T& z)
+    {
+	typedef typing::decplx_t<T> RT;
+	switch ( std::fpclassify(z.imag()) )
+	{
+	    case FP_NORMAL :
```

Comment:
Is this an intentional fall-through?  Usually good practice to mark with an explicit `/* fall-through */` comment